### DISCUSSION TITLE
Release Notes in Renovate pull request is wrong if multiple dependencies are hosted in the same GitHub Repository

### DISCUSSION THREAD

**Discussion by suzuki-shunsuke** [renovatebot/renovate#21746] [Request Help]
### How are you running Renovate?

Mend Renovate hosted app on github.com

### If you're self-hosting Renovate, select which platform you are using.

None

### Was this something which used to work for you, and then stopped?

It used to work, and then stopped

### Wanted end result.

Reproduction code: https://github.com/suzuki-shunsuke/reproduction-code-renovate-monorepo

renovate.json

```json
{
   "branchConcurrentLimit": 0,
   "extends": [
      "config:base"
   ],
   "prConcurrentLimit": 0,
   "prHourlyLimit": 0,
   "regexManagers": [
      {
         "datasourceTemplate": "github-releases",
         "extractVersionTemplate": "^bar_(?<version>v\\d+\\.\\d+\\.\\d+(?:-\\d+)?)$",
         "fileMatch": [
            "^foo\\.txt$"
         ],
         "matchStrings": [
            "bar: (?<currentValue>v\\d+\\.\\d+\\.\\d+)"
         ],
         "packageNameTemplate": "suzuki-shunsuke/reproduction-code-renovate-monorepo",
         "depNameTemplate": "suzuki-shunsuke/reproduction-code-renovate-monorepo/bar"
      },
      {
         "datasourceTemplate": "github-releases",
         "extractVersionTemplate": "^foo_(?<version>v\\d+\\.\\d+\\.\\d+(?:-\\d+)?)$",
         "fileMatch": [
            "^foo\\.txt$"
         ],
         "matchStrings": [
            "foo: (?<currentValue>v\\d+\\.\\d+\\.\\d+)"
         ],
         "packageNameTemplate": "suzuki-shunsuke/reproduction-code-renovate-monorepo",
         "depNameTemplate": "suzuki-shunsuke/reproduction-code-renovate-monorepo/foo"
      }
   ]
}
```

foo.txt

```
foo: v0.1.0
bar: v0.1.0
```

Renovate created two pull requests.

- https://github.com/suzuki-shunsuke/reproduction-code-renovate-monorepo/pull/7
- https://github.com/suzuki-shunsuke/reproduction-code-renovate-monorepo/pull/8

## Expected behaviour

The release note in https://github.com/suzuki-shunsuke/reproduction-code-renovate-monorepo/pull/7 has two links to GitHub Releases.

- [foo_v0.1.1](https://github.com/suzuki-shunsuke/reproduction-code-renovate-monorepo/releases/tag/foo_v0.1.1)
- [foo_v0.1.2](https://github.com/suzuki-shunsuke/reproduction-code-renovate-monorepo/releases/tag/foo_v0.1.2)

And the release note in https://github.com/suzuki-shunsuke/reproduction-code-renovate-monorepo/pull/8 has two links to GitHub Releases.

- [bar_v0.1.1](https://github.com/suzuki-shunsuke/reproduction-code-renovate-monorepo/releases/tag/bar_v0.1.1)
- [bar_v0.1.2](https://github.com/suzuki-shunsuke/reproduction-code-renovate-monorepo/releases/tag/bar_v0.1.2)

## Actual behaviour

The release note in https://github.com/suzuki-shunsuke/reproduction-code-renovate-monorepo/pull/7 has two links.

- [foo_v0.1.1...foo_v0.1.2](https://togithub.com/suzuki-shunsuke/reproduction-code-renovate-monorepo/compare/foo_v0.1.1...foo_v0.1.2)
- [bar_v0.1.0...bar_v0.1.1](https://togithub.com/suzuki-shunsuke/reproduction-code-renovate-monorepo/compare/bar_v0.1.0...bar_v0.1.1) ⚠️ This should compare foo_v0.1.1 with foo_v0.1.0

And the release note in https://github.com/suzuki-shunsuke/reproduction-code-renovate-monorepo/pull/8 has two links.

- [foo_v0.1.1...foo_v0.1.2](https://togithub.com/suzuki-shunsuke/reproduction-code-renovate-monorepo/compare/foo_v0.1.1...foo_v0.1.2)
- [bar_v0.1.0...bar_v0.1.1](https://togithub.com/suzuki-shunsuke/reproduction-code-renovate-monorepo/compare/bar_v0.1.0...bar_v0.1.1) ⚠️ This should compare bar_v0.1.1 with bar_v0.1.0

Tw things are unexpected.

- The link is not GitHub Release but comparison
- Tags of the comparison are wrong.

### What you tried so far.

I have no idea.

### Relevant debug logs

<details><summary>Logs</summary>

```
Copy/paste the relevant log(s) here, between the starting and ending backticks
```

</details>

**Comment by rarkins** [comment]
This only happens if there are multiple links in the same PR?

**Reply by suzuki-shunsuke** [reply]
I'm not sure, but the comparison links in the following pull requests are correct.

- https://github.com/suzuki-shunsuke/reproduction-code-renovate-monorepo/pull/2
- https://github.com/suzuki-shunsuke/reproduction-code-renovate-monorepo/pull/4

But I expect links are not commits between versions but GitHub Releases because the datasource is `github-releases`.
